nodes/monitor.go

```diff
@@ -91,6 +91,8 @@ func (mon *NodeMonitor) loop() {
 	}
 }
 
+var headListCache []int
```

Comment:
Ah, actually, I think there's a cornercase here. 

What if.. We have a typical recent reorg, so 3 nodes are at `13M+1`, one of them has a different head (or even different head-1). So there's a fork at `13M`. Now, we just found out that an old geth version forked 100 blocks back, so we connect it to the nodemon. 

The algo will now see 'yep, this differs at block `13M`', and stop there, and miss that the fork occurred 100 blocks earlier. 

Same if we hook it up to an ETC node, or a ropsten node, which forked off at genesis.

Reply:
You could probably salvage this idea if instead of caching only the numbers, you cache the `(number,hash)`. Then you could just check, for each number, "is this node on an already known fork?".

Reply:
Yeah I'm aware of that case, I would've solved it by deleting the cache every now and then, but yeah maybe storing the hash might be a better idea. headListCache is badly chosen, I can rename it

Reply:
Well, also there's the fact that right now, we don't support adding nodes without shutting the thing down, so it can't really happen right now ( -- but that'll probably change at some point, and also it's possible that the 'old' node is configured but not available the first time it runs)

Reply:
Another idea: you could keep the numbers-only, but always run the binary search on the _lower_ portions first. Say you had a fork at 30, another at 50, and are now at 100. First a binary search [0,30]. If nothing, [30, 50], then [50, head ( 100)]. 

It will do more binary searches, but each binary search will just revisit the same indices over and over again, and those headers will be cached, so won't actually make a network request (until we clear the cache).

Reply:
There's a one-statement fix for this issue